File: 1657-find-the-winner-of-an-array-game/1657-find-the-winner-of-an-array-game.java
class Solution {
    public int getWinner(int[] arr, int k) {
        int m=arr[0];
        int c=0;
        for(int i=1;i<arr.length;i++){
            if(m<arr[i]) {
                c=0;
                m=arr[i];
            }
            c+=1;
            if (c==k) return m;
        }
        return m;

    }
}

File: 1657-find-the-winner-of-an-array-game/1657-find-the-winner-of-an-array-game.py
class Solution:
    def getWinner(self, arr: List[int], k: int) -> int:
        m=arr[0]
        d=defaultdict(int)
        for i in range(1,len(arr)):
            if m<arr[i]:
                m=arr[i]
            d[m]+=1
            if d[m]==k:return m
        print(d)        
        return m         

                
        